pkg/crc/errors/serialerror.go

```diff
@@ -0,0 +1,22 @@
+package errors
+
+import "encoding/json"
+
+func EventuallySerializableError(err error) *SerializableError {
```

Comment:
`errors.EventuallySerializableError` can be also:
* `errors.EventuallySerializable` (remove the extra Error)
* `errors.MaybeSerializable`

Otherwise, lgtm!

Reply:
`errorMessage` naming did not indicate the `maybe`/`eventually`/... Imo it's more readable without this hint, hopefully it makes sense to assume `errors.EventuallySerializableError(nil)` returns `nil`? In which case, this could become `MakeSerializable`, `ToSerializable` (feels a bit weird without the `Error` suffix, so maybe keep it too).

Reply:
I changed it to `ToSerializableError` make it much readable and also provide sense that it will serialize the error.